.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,92 @@
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+     - v**
+  pull_request:
+    branches: [ master ]
+     
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+      
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+      
+    - name: Log into registry ${{ env.REGISTRY }}
+      if: github.event_name == 'push'
+      uses: docker/login-action@v2.1.0
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Docker Setup QEMU
+      uses: docker/setup-qemu-action@v2.1.0
```

Comment:
I assume we need qemu for the docker build? Or?

Also, it looks like `setup-qemu-action` takes an optional list of platforms to target. Can we make the build process more light-weight by restricting the platforms to the ones we actually use?

Reply:
The qemu step is used by buildx to build docker images for other platforms. We are targeting amd64, 32 bit arm, and 64 bit arm in the docker build step. I don't think targeting platforms here will save us much since this task only takes a few seconds to run.

Reply:
That all makes sense - thanks.